src/workerd/api/trace.h

```diff
@@ -209,6 +211,25 @@ class TraceItem::QueueEventInfo final: public jsg::Object {
   const Trace::QueueEventInfo& eventInfo;
 };
 
+class TraceItem::EmailEventInfo final: public jsg::Object {
+public:
+  explicit EmailEventInfo(kj::Own<Trace> trace, const Trace::EmailEventInfo& eventInfo);
+
+  kj::StringPtr getMailFrom();
+  kj::StringPtr getRcptTo();
+  uint32_t getRawSize();
+
+  JSG_RESOURCE_TYPE(EmailEventInfo) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(mailFrom, getMailFrom);
+    JSG_READONLY_PROTOTYPE_PROPERTY(rcptTo, getRcptTo);
+    JSG_READONLY_PROTOTYPE_PROPERTY(rawSize, getRawSize);
+  }
+
+private:
+  kj::Own<Trace> trace;
+  const Trace::EmailEventInfo& eventInfo;
```

Comment:
I don't believe this is a new problem at all but I do generally have a concern about using const refs in jsg::Objects like this. What guarantee do we have that the ref is going to stay alive for as long as the JS wrapper? For instance, if someone happens to store the info object at global scope, for instance, what happens when the properties are accessed?

Reply:
I agree, we should look at this.